### ISSUE TITLE
prefer-template should fail if only variables are passed to plus operator.

### ISSUE THREAD

**Issue by CraigHarley** [eslint/eslint#10734]
**Tell us about your environment**

* **ESLint Version:v4.18.1**
* **Node Version:v10.7.0**
* **npm Version:6.1.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    env: {
        browser: true,
        commonjs: true,
        es6: true
    },
    extends: [
        "eslint:recommended",
        "plugin:vue/recommended",
        "plugin:flowtype/recommended"
    ],
    parser: "vue-eslint-parser",
    parserOptions: {
        parser: "babel-eslint",
        sourceType: "module"
    },
    plugins: [
        "vue",
        "flowtype-errors",
        "flowtype"
    ],
    rules: {
        indent: [2, 4],
        quotes: [2, "double"],
        semi: [2, "always"],
        "arrow-body-style": [2, "as-needed"],
        "no-unused-vars": [2, {
            vars: "all",
            args: "after-used",
            ignoreRestSiblings: false
        }
        ],
        "no-trailing-spaces": 2,
        eqeqeq: 2,
        "no-console": 2,
        "prefer-template": 2,
        "prefer-const": 2,
        "vue/require-v-for-key": "off",
        "vue/no-parsing-error": "off",
        "vue/return-in-computed-property": "off",
        "linebreak-style": "off",
        "comma-dangle": [2, "never"],
        "quote-props": [2, "as-needed"],
        "eol-last": [2, "always"],
        "flowtype-errors/show-errors": 2,
        "vue/name-property-casing": [2, "kebab-case"],
        "vue/html-self-closing": [2, {
            html: {
                void: "never",
                normal: "never",
                component: "never"
            }
        }],
        "new-parens": 2,
        "object-curly-spacing": [2, "always"],

        // these rules can be turned on later, they can't be auto-fixed.
        "vue/html-indent": "off",
        "vue/max-attributes-per-line": "off",
        "vue/attributes-order": "off",
        "vue/require-prop-types": "off",
        "vue/require-default-prop": "off",
        "vue/no-confusing-v-for-v-if": "off",
        "vue/this-in-template": "off"
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
	var first = '123';
	var second = '456';
	var concat1 = first + second; // I would expect this to fail prefer-template, since it's ambiguous, first and second could be numbers. 
	var concat2 = first + second + '789'; // This fails

```

```bash
eslint ./src/**/*.{js,vue}
```

**What did you expect to happen?**

I would expect `var concat1 = first + second;` to fail `prefer-template`, or at least to be able to configure eslint to fail in this case.

**What actually happened? Please include the actual, raw output from ESLint.**
```
yarn run v1.5.1
$ eslint ./src/**/*.{js,vue}
Done in 6.54s.
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue.

I'm not sure I understand what you're proposing. I agree that using the `+` operator with two variables is "ambiguous" since the variables could be numbers. However, could you clarify why would expect `prefer-template` to report an error as a result? To me, it seems like it only makes sense for `prefer-template` to report an error when the code is unambiguously a string concatenation, because otherwise the rule would incorrectly report some problems in cases where numeric addition is intended.

**Comment by CraigHarley** [comment]
Yeah that's fair. It seems like this is a limitation somewhat of not using a static type checker.

I'll close this issue.
